rust/extractor/src/translate/base.rs

```diff
@@ -298,20 +298,18 @@ impl<'a> Translator<'a> {
         if let Some(value) = semantics
             .hir_file_for(expanded)
             .macro_file()
-            .and_then(|macro_file| {
-                semantics
-                    .db
-                    .parse_macro_expansion_error(macro_file.macro_call_id)
-            })
+            .and_then(|macro_call_id| semantics.db.parse_macro_expansion_error(macro_call_id))
         {
             if let Some(err) = &value.err {
                 let error = err.render_to_string(semantics.db);
-
-                if err.span().anchor.file_id == semantics.hir_file_for(node.syntax()) {
+                let hir_file_id = semantics.hir_file_for(node.syntax());
+                if Some(err.span().anchor.file_id.file_id())
+                    == hir_file_id.file_id().map(|f| f.file_id(semantics.db))
```

Comment:
how did you figure out that you needed the file_id of the file_id of the file_id of file_id ?  ;-)